CMakeLists.txt

```diff
@@ -646,56 +647,58 @@ if (BUILD_SHARED_LIBS)
-set(includedir "${CMAKE_INSTALL_FULL_INCLUDEDIR}")
-set(PACKAGE "${PROJECT_NAME}")
-set(VERSION "${FULL_VERSION}")
-set(SDL_VERSION "${SDL_REQUIRED_VERSION}")
-string(JOIN " " PC_REQUIRES ${PC_REQUIRES})
-configure_file(SDL2_image.pc.in SDL2_image.pc @ONLY)
-
-install(EXPORT SDL2ImageExports NAMESPACE SDL2::
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/SDL2_image" FILE SDL2_image-targets.cmake)
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/SDL2_image-config.cmake" DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/SDL2_image")
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/SDL2_image.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
-install(FILES "LICENSE.txt" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/licenses/${PROJECT_NAME}")
-if (NOT (WIN32 OR CYGWIN OR MINGW))
-    if(BUILD_SHARED_LIBS)
-        set(SOEXT ${CMAKE_SHARED_LIBRARY_SUFFIX}) # ".so", ".dylib", etc.
-        get_target_property(SONAME SDL2_image OUTPUT_NAME)
-        if(NOT ANDROID)
-            install(CODE "
-				execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink
-					\"lib${SONAME}${SOEXT}\" \"libSDL2_image${SOEXT}\"
-					WORKING_DIRECTORY \"${PROJECT_BINARY_DIR}\")")
-            install(FILES ${PROJECT_BINARY_DIR}/libSDL2_image${SOEXT} DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+    if (INSTALL_EXTRA_TARGETS)
+        install(TARGETS ${INSTALL_EXTRA_TARGETS} EXPORT SDL2ImageExports
+            ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+            RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+            PUBLIC_HEADER DESTINATION "${CMAKE_CURRENT_BINARY_DIR}/external_include"
+            )
+    endif()
+
+    configure_package_config_file(SDL2_image-config.cmake.in SDL2_image-config.cmake
+        INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/SDL2_image")
+
+    set(prefix "${CMAKE_INSTALL_PREFIX}")
+    set(exec_prefix "\${prefix}")
+    set(libdir "\${exec_prefix}/${CMAKE_INSTALL_LIBDIR}")
+    set(includedir "\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}")
+    set(PACKAGE "${PROJECT_NAME}")
+    set(VERSION "${FULL_VERSION}")
+    set(SDL_VERSION "${SDL_REQUIRED_VERSION}")
+    string(JOIN " " PC_REQUIRES ${PC_REQUIRES})
+    file(GENERATE OUTPUT SDL2_image.pc INPUT SD2_image.pc.in)
+
+    install(EXPORT SDL2ImageExports NAMESPACE SDL2_image::
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/SDL2_image" FILE SDL2_image-targets.cmake)
+    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/SDL2_image-config.cmake" DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/SDL2_image")
+    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/SDL2_image.pc" DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+    install(FILES "LICENSE.txt" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/licenses/${PROJECT_NAME}")
+    if (NOT (WIN32 OR CYGWIN OR MINGW))
+        if(BUILD_SHARED_LIBS)
+            set(SOEXT ${CMAKE_SHARED_LIBRARY_SUFFIX}) # ".so", ".dylib", etc.
+            get_target_property(SONAME SDL2_image OUTPUT_NAME)
+            if(NOT ANDROID)
```

Comment:
Should this go into the (NOT WIN32...) check above?

Reply:
This code creates a dynamic link e.g. `libSDL2_image.so` -> `libSDL2_image-2.0.so` and is only needed for shared libraries.
The output name with versioning suffix `-2.0` is only applied to shared libraries, static libraries are always called `libSDL2_image.a` without versioning suffix. SDL2 has the same behavior.

What I don't understand is why android does not need the dynamic link,
but it does not matter if people are using our FindSDL2Image module.